api/v1/linstorsatelliteconfiguration_types.go

```diff
@@ -47,6 +47,12 @@ type LinstorSatelliteConfigurationSpec struct {
 	// +patchMergeKey=name
 	// +patchStrategy=merge
 	Properties []LinstorNodeProperty `json:"properties,omitempty"`
+
+	// InternalTLS configures secure communication for the LINSTOR Satellite.
+	//
+	// If set, the control traffic between LINSTOR Controller and Satellite will be encrypted using mTLS.
+	// +kubebuilder:validation:Optional
+	InternalTLS *TLSConfig `json:"internalTLS,omitempty"`
```

Comment:
Does this one not need `+nullable`?

Reply:
Not in my testing. `nullable` seems to be only needed because we use "server-side apply" and at some point the API server converts a `internalTLS: {}` into `internalTLS: null`, which then triggers validation. I only added it to the satellite because that is the one point where the operator can trigger this behaviour.

Thinking about it some more, we could also add it here and in the LinstorCluster CRD, should any of our users also use server-side apply for their own configuration :thinking: